### PR TITLE
fixes #11923 [backport:1.4]

### LINKED ISSUE #11923: hasCustomPragma fails on nnkVarTy/nnkBracketExpr nodes
So I've looked into the problem I was trying to fix in #11879 a little more closely and found another case where even my PR doesn't fix the problem. I think the root cause is that ``getImpl`` does not work when a ``nnkVarTy`` or a ``nnkBracketExpr`` are passed to it, this is something that `customPragmaNode` tries to work around.

### Example

```nim
import macros

type
  MyObj[T] = object
    test: Point[T]

  Point[T] = object
    x, y: T

template testPragma {.pragma.}

proc case1(x: var MyObj) =
  for k, v in fieldPairs(x):
    when v.hasCustomPragma(testPragma):
      echo("hi")

proc case2(x: MyObj) =
  for k, v in fieldPairs(x):
    when v.hasCustomPragma(testPragma):
      echo("hi")

var x = MyObj[float](test: Point[float](x: 1.3, y: 0.0))
case2(x)
case1(x)
```

### Current Output
```
stack trace: (most recent call last)
/Users/dom/projects/nim/lib/core/macros.nim(1538, 36) hasCustomPragma
/Users/dom/projects/nim/lib/core/macros.nim(1478, 25) customPragmaNode
/private/tmp/aporia/a1.nim(23, 6) template/generic instantiation of `case2` from here
/private/tmp/aporia/a1.nim(19, 11) template/generic instantiation of `hasCustomPragma` from here
/Users/dom/projects/nim/lib/core/macros.nim(1478, 25) Error: node is not a symbol
```

### Expected Output

No error. ``hasCustomPragma`` returns false.

### Possible Solution

A variant of #11879 or changing ``getImpl`` to support ``nnkVarTy``/``nnkBracketExpr``

### Additional Information

```
$ nim -v
Nim Compiler Version 0.20.99 [MacOSX: amd64]
Compiled at 2019-08-06
Copyright (c) 2006-2019 by Andreas Rumpf

git hash: 8407a574992ebd6bccec647a902cf54a4de8db18
active boot switches: -d:release
```

### PR DESCRIPTION
**Pull request by Dankr4d**
fixes #11923

### CODE REVIEW

**Comment by ringabout** [comment]
please add a test

**Comment by Dankr4d** [comment]
Should I rebase both commits into a single commit?

**Comment by timotheecour** [comment]
> Should I rebase both commits into a single commit?

in general it's not needed, because this repo is setup to enforce squash and merge of PRs, so keeping the individual commits is harmless and maintains history (if preserving individual commit history has value, which is up to you to judge)

File: lib/core/macros.nim [hunk 1]
```diff
@@ -1524,7 +1529,18 @@ proc customPragmaNode(n: NimNode): NimNode =
   if n.kind in {nnkDotExpr, nnkCheckedFieldExpr}:
     let name = $(if n.kind == nnkCheckedFieldExpr: n[0][1] else: n[1])
     let typInst = getTypeInst(if n.kind == nnkCheckedFieldExpr or n[0].kind == nnkHiddenDeref: n[0][0] else: n[0])
-    var typDef = getImpl(if typInst.kind == nnkVarTy: typInst[0] else: typInst)
+
+    var typDef: NimNode
+    if typInst.kind == nnkVarTy:
+      if typInst[0].kind == nnkBracketExpr:
+        typDef = getImpl(typInst[0][0])
```
Comment (Araq) [inline lib/core/macros.nim:1536]: Dubious. How do we know that we don't skip over useful information?

**Comment by Araq** [comment]
Ping.

**Comment by Dankr4d** [comment]
I'm currently more or less busy and didn't get it into reviewing the code. I'll have a look into it and check your remark.

**Comment by ringabout** [comment]
#11923 seems to be sovled by https://github.com/nim-lang/Nim/issues/11923.

**Pull request closed without merging**